test/sv/errors.mlir

```diff
@@ -0,0 +1,14 @@
+// RUN: circt-opt %s -split-input-file -verify-diagnostics
+
+rtl.module @test_instance_exist_error() {
+  // expected-error @+1 {{Symbol not found: @noexist.}}
+  %b = sv.interface.instance : !sv.interface<@noexist>
+}
+
+// -----
+
+module {
+  rtl.module @foo () {  }
+  // expected-error @+1 {{Symbol @foo is not an InterfaceOp.}}
+  %b = sv.interface.instance : !sv.interface<@foo>
+}
```

Comment:
maybe add a test for this case as well?

https://github.com/llvm/circt/pull/294/files#diff-08f0541c97d71a4bc0a80bd733f367f0d4df5210011b76e3772ea7b0c5c39e2eR119

Reply:
What region is not a symbol table?

Reply:
and doesn't expose the symbol table above? (Is that a thing in MLIR?)

Reply:
I guess I was thinking at the very top level, not inside an MLIR ModuleOp (which isn't [necessarily](https://github.com/llvm/llvm-project/commit/00c6ef8628a6ee062d5104692a92d5e000dd7c05#diff-e0acb49d6ec4a97460a1af5fe8f80abc39f04d172d6c9a08c33da1209115cd4b) [required](https://github.com/llvm/llvm-project/commit/b57980309a699a47a46b5b5749b36fea06eaaf33#diff-e0acb49d6ec4a97460a1af5fe8f80abc39f04d172d6c9a08c33da1209115cd4b)). I guess in practice this won't be an issue. For instance, `rtl.module` [ensures](https://github.com/llvm/circt/blob/master/include/circt/Dialect/RTL/Structure.td#L3) it is within a ModuleOp.

If this case can't be hit by IR that otherwise passes verification, maybe it should be an assertion?

Reply:
When I put it at the top level, it finds a symbol table since MLIR implicitly constructs a module around any top level operations!

> If this case can't be hit by IR that otherwise passes verification, maybe it should be an assertion?

I'm pretty sure an IR can be written which cannot locate a symbol table and I just don't know what it is or whether or not it's feasible for me to do so. I'm going to keep it as-is in the interest of safety.

Reply:
> MLIR implicitly constructs a module around any top level operations!

Agreed, but patches landed just this afternoon that are removing implicit assumptions about ModuleOp in the parser and pass manager, and I think this won't always be the case.

> I'm going to keep it as-is in the interest of safety. 

Anyway, sounds good to keep it for now. It's a good check but no need to bend over backwards to test a weird case we shouldn't ever see